editor/editor_inspector.cpp

```diff
@@ -2696,638 +2867,656 @@ void EditorInspector::update_tree() {
-						(!filter.is_empty() || !restrict_to_basic || (N->get().usage & PROPERTY_USAGE_EDITOR_BASIC_SETTING))) {
-					break;
+				if (!show_categories) {
+					continue;
 				}
-				if (N->get().usage & PROPERTY_USAGE_CATEGORY) {
-					valid = false;
-					break;
+
+				// Hide the "MultiNodeEdit" category for MultiNodeEdit.
+				if (Object::cast_to<MultiNodeEdit>(object) && p.name == "MultiNodeEdit") {
+					continue;
 				}
-				N = N->next();
-			}
-			if (!valid) {
-				continue; // Empty, ignore it.
-			}
 
-			// Create an EditorInspectorCategory and add it to the inspector.
-			EditorInspectorCategory *category = memnew(EditorInspectorCategory);
-			main_vbox->add_child(category);
-			category_vbox = nullptr; //reset
-
-			String type = p.name;
-			String label = p.name;
-			doc_name = p.name;
-
-			// Use category's owner script to update some of its information.
-			if (!EditorNode::get_editor_data().is_type_recognized(type) && p.hint_string.length() && ResourceLoader::exists(p.hint_string)) {
-				Ref<Script> scr = ResourceLoader::load(p.hint_string, "Script");
-				if (scr.is_valid()) {
-					StringName script_name = EditorNode::get_editor_data().script_class_get_name(scr->get_path());
-
-					// Update the docs reference and the label based on the script.
-					Vector<DocData::ClassDoc> docs = scr->get_documentation();
-					if (!docs.is_empty()) {
-						// The documentation of a GDScript's main class is at the end of the array.
-						// Hacky because this isn't necessarily always guaranteed.
-						doc_name = docs[docs.size() - 1].name;
-					}
-					if (script_name != StringName()) {
-						label = script_name;
+				// Iterate over remaining properties. If no properties in category, skip the category.
+				List<PropertyInfo>::Element *N = E_property->next();
+				bool valid = true;
+				while (N) {
+					if (!N->get().name.begins_with("metadata/_") && N->get().usage & PROPERTY_USAGE_EDITOR &&
+							(!filter.is_empty() || !restrict_to_basic || (N->get().usage & PROPERTY_USAGE_EDITOR_BASIC_SETTING))) {
+						break;
 					}
-
-					// Find the icon corresponding to the script.
-					if (script_name != StringName()) {
-						category->icon = EditorNode::get_singleton()->get_class_icon(script_name, "Object");
-					} else {
-						category->icon = EditorNode::get_singleton()->get_object_icon(scr.ptr(), "Object");
+					if (N->get().usage & PROPERTY_USAGE_CATEGORY) {
+						valid = false;
+						break;
 					}
+					N = N->next();
+				}
+				if (!valid) {
+					continue; // Empty, ignore it.
 				}
-			}
 
-			if (category->icon.is_null() && !type.is_empty()) {
-				category->icon = EditorNode::get_singleton()->get_class_icon(type, "Object");
-			}
+				// Create an EditorInspectorCategory and add it to the inspector.
+				EditorInspectorCategory *category = memnew(EditorInspectorCategory);
+				main_vbox->add_child(category);
+				category_vbox = nullptr; //reset
```

Comment:
```suggestion
				category_vbox = nullptr; // Reset.
```

Reply:
This is out of the scope of my changes, I just placed an `if` enclosing it, it's only "white space changes".

Reply:
The code is updated so it's best to get it in line, and not much work, so please update it anyway :)